tests/update_workflow_test.go

```diff
@@ -289,6 +301,15 @@ func (s *functionalSuite) TestUpdateWorkflow_NewSpeculativeWorkflowTask_AcceptCo
 			s.EqualValues(tv.String("success-result", "1"), decodeString(s, updateResult.GetOutcome().GetSuccess()))
 			s.EqualValues(0, updateResp.ResetHistoryEventId)
 
+			// Test non-blocking poll
+			for _, waitPolicy := range []*updatepb.WaitPolicy{{LifecycleStage: enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_UNSPECIFIED}, nil} {
```

Comment:
I would test an incomplete async update to verify that this is indeed non-blocking.

Reply:
IMO there is already coverage for that -- I made a comment inline here https://github.com/temporalio/temporal/pull/5114/files?show-viewed-files=true&file-filters%5B%5D=#r1396421932